### PR TITLE
Adding tf2_kdl tests

### PR DESCRIPTION
**Pull request by ahcorde**

### CODE REVIEW

File: tf2_kdl/test/test_tf2_kdl.cpp [hunk 1]
```diff
@@ -109,23 +105,26 @@ TEST(TfKDL, ConvertVector)
...
-  t.header.stamp = builtin_interfaces::msg::Time(2.0);
+  t.header.stamp.sec = 2.0;
+  t.header.stamp.nanosec = 0.0;
   t.header.frame_id = "A";
   t.child_frame_id = "B";
   tf_buffer->setTransform(t, "test");
 
   int retval = RUN_ALL_TESTS();
-  delete tf_buffer;
+  // // delete tf_buffer;
```
Comment (vmayoral) [inline tf2_kdl/test/test_tf2_kdl.cpp:128]: @ahcorde why are we keeping this?

**Comment by vmayoral** [comment]
Also, this work overrides https://github.com/ros2/geometry2/pull/90/files. Can you comment on what's your view?

File: tf2_kdl/CMakeLists.txt [hunk 2]
```diff
@@ -28,26 +20,26 @@ install(PROGRAMS scripts/test.py
...
-#   if(TARGET tests)
-#     add_dependencies(tests test_kdl)
-#   endif()
-# endif()
+if(BUILD_TESTING)
+	find_package(ament_cmake_gtest REQUIRED)
+
+  ament_add_gtest(test_kdl test/test_tf2_kdl.cpp)
+
+	target_include_directories(test_kdl PUBLIC
```
Comment (vmayoral) [inline tf2_kdl/CMakeLists.txt:28]: The approach you follow here is different from what's been used at https://github.com/ros2/geometry2/pull/90/files#diff-61737c5a9ef69ef2118eafbc59732b6aR45

File: tf2_kdl/include/tf2_kdl/tf2_kdl.h [hunk 3]
```diff
@@ -108,9 +112,9 @@ geometry_msgs::PointStamped toMsg(const tf2::Stamped<KDL::Vector>& in)
  * \param out The point converted to a timestamped KDL Vector.
  */
 inline
-void fromMsg(const geometry_msgs::PointStamped& msg, tf2::Stamped<KDL::Vector>& out)
+void fromMsg(const geometry_msgs::msg::PointStamped& msg, tf2::Stamped<KDL::Vector>& out)
 {
-  out.stamp_ = msg.header.stamp;
+  out.stamp_ = tf2::TimePoint(durationFromSec(msg.header.stamp.sec + msg.header.stamp.nanosec/NSEC_PER_SECOND));
```
Comment (vmayoral) [inline tf2_kdl/include/tf2_kdl/tf2_kdl.h:117]: You apply here a simplified version of what's been used at https://github.com/ros2/geometry2/pull/90/files#diff-2aeab99ef52e795be884c1fbbef16e44R195 where the function used was defined at https://github.com/ros2/geometry2/pull/90/files#diff-2aeab99ef52e795be884c1fbbef16e44R78.

This needs to get repeated over a few places so IMHO, creating a function sounds more appropriate.

**Comment by vmayoral** [comment]
Moreover, branch fails to compile:

```bash
tarting >>> tf2_kdl
--- stderr: tf2_kdl
In file included from /Users/victor/ros2_ws/src/ros2/geometry2/tf2_kdl/test/test_tf2_kdl.cpp:33:
/Users/victor/ros2_ws/install/include/tf2_kdl/tf2_kdl.h:37:10: fatal error: 'geometry_msgs/PointStamped.h' file not found
#include <geometry_msgs/PointStamped.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/test_kdl.dir/test/test_tf2_kdl.cpp.o] Error 1
make[1]: *** [CMakeFiles/test_kdl.dir/all] Error 2
make: *** [all] Error 2
---
Failed   <<< tf2_kdl	[ Exited with code 2 ]
```

**Comment by vmayoral** [comment]
@ahcorde please double check https://github.com/ros2/geometry2/pull/90

File: tf2_kdl/test/test_tf2_kdl.cpp (hunk 1 above)
Comment (ahcorde) [reply tf2_kdl/test/test_tf2_kdl.cpp:128]: my bad

**Comment by ahcorde** [comment]
I don't know wwhy is failling because I'm not including :
```
#include <geometry_msgs/PointStamped.h>
```
You see here that this line is not the same:
https://github.com/AcutronicRobotics/geometry2/blob/fix-kdl/tf2_kdl/include/tf2_kdl/tf2_kdl.h#L37

Are you in the right branch?

**Comment by vmayoral** [comment]
Changes here and also at https://github.com/AcutronicRobotics/geometry2/commits/fix-test-tf2_kdl have been added to https://github.com/ros2/geometry2/pull/90 manually.

Moreover all these changes have been merged into [master branch](https://github.com/AcutronicRobotics/geometry2) (so that CI picks them).

Closing here then.

**Pull request closed without merging**